Tests/Realm.Tests/Sync/SyncTestHelpers.cs

```diff
@@ -167,28 +164,36 @@ public static (string[] RemainingArgs, IDisposable? Logger) SetLoggerFromArgs(st
 
         private static async Task CreateBaasAppsAsync()
         {
-            if (_apps[AppConfigType.Default].AppId != string.Empty || BaasUri == null)
+            if (_apps[AppConfigType.Default].AppId != string.Empty || (BaasUri == null && _baaSaasApiKey == null))
             {
                 return;
             }
 
 #if !UNITY
-            try
-            {
-                var cluster = ConfigHelpers.GetSetting("Cluster")!;
-                var apiKey = ConfigHelpers.GetSetting("ApiKey")!;
-                var privateApiKey = ConfigHelpers.GetSetting("PrivateApiKey")!;
-                var groupId = ConfigHelpers.GetSetting("GroupId")!;
-                var differentiator = ConfigHelpers.GetSetting("Differentiator") ?? "local";
+            var cluster = ConfigHelpers.GetSetting("Cluster")!;
```

Comment:
I'm a little bit conflicted if it makes sense to add a "Mode" to the config file, instead of relying on the fields in the config file to implicitly deduce the kind of baas client we need.